gr-iio/lib/iio_priv.h

```diff
@@ -0,0 +1,29 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2023 Analog Devices Inc.
+ * Author: Adrian Suciu <adrian.suciu@analog.com>
+
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifndef INCLUDED_IIO_IIO_PRIV_H
+#define INCLUDED_IIO_IIO_PRIV_H
+
+#include <iio.h>
+
+namespace gr {
+namespace iio {
+constexpr unsigned int channel_size(const iio_channel* chn)
```

Comment:
Ah, shoot, my bad, if you do it this way: because `iio_channel_get_data_format` is C and not C++, it can't be `constexpr`, so this whole function cannot be `constexpr`, which MSVC recognizes and hence fails to compile this.

This might not be happening on other platforms since iio.h declares the function `__attribute__(const)` there; I don't think there's an MSVC equivalent to that.

Now, what is a good way forward? Hard one: probably start by dropping the `constexpr` here (annoying, I know).

Maybe upstream fix libiio/iio.h to define their `__cnst` to evaluate to `constexpr` in case of `__cplusplus`, and then add a version check so that GNU Radio uses `constexpr` for `channel_size()` when the system's libiio allows for that.

Speaking of version checks: Note that iio.h doesn't have a `#define LIBIIO_VERSION` or similar, which makes making code compatible unnecessarily hard on people that work with libiio a lot :-)

Reply:
I removed the constexpr and pushed .. 
I agree with LIBIIO_VERSION in iio.h (especially now that we're moving to libiio v1)
Regarding constexpr ? I'm a little confused, wouldn't __cnst be a macro in GR and used in iio_priv.h ?